src/wallet/rpcwallet.cpp

```diff
@@ -1133,6 +1133,9 @@ UniValue ListReceived(const UniValue& params, bool fByAccounts)
     if (params.size() > 0)
         nMinDepth = params[0].get_int();
 
+    bool fFilterAddress = !fByAccounts && params.size() > 3;
+    const CBitcoinAddress filterAddress = fFilterAddress ?  CBitcoinAddress(params[3].get_str()) : CBitcoinAddress{};
+
```

Comment:
I think it would be clearer to move these two lines down a bit so the params are being tested in order. I also have a slight preference for an `if params.size() > 3` block to match the existing tests.